src/libYARP_os/src/yarp/os/Stamp.cpp

```diff
@@ -106,8 +107,7 @@ bool yarp::os::Stamp::write(ConnectionWriter& connection) const
 
 int yarp::os::Stamp::getMaxCount() const
 {
-    // a very conservative maximum
-    return 32767;
+    return std::numeric_limits<int>::max();
```

Comment:
The sequence number is transmitted on the network as `int32_t`, so I think it make sense to reflect this in the maxCount .

Reply:
Even if indeed I don't think there are any platforms that we support for which int is not 32-bit .

Reply:
See https://en.wikipedia.org/wiki/64-bit_computing#64-bit_data_models .

Reply:
So do you think that it is better to use `std::numeric_limits<int32_t>::max()`?

Reply:
I agree, but probably the whole stamp class should be using int32_t.
Actually it should probably use an unsigned type...

Reply:
I opened #2365, let's merge as it is now, and fix it everywhere later.